src/infrastructure/api/index.ts

```diff
@@ -0,0 +1,112 @@
+import AWS from 'aws-sdk';
+import express, { Request } from 'express';
+import mysql from 'mysql2/promise';
+import vehicleQueryFunctionFactory from '../../app/vehicleQueryFunctionFactory';
+import testQueryFunctionFactory from '../../app/testQueryFunctionFactory';
+import { getResultsDetails, getVehicleDetails } from '../../domain/enquiryService';
+import ParametersError from '../../errors/ParametersError';
+import ResultsEvent from '../../interfaces/ResultsEvent';
+import VehicleEvent from '../../interfaces/VehicleEvent';
+import DatabaseService from '../databaseService';
+import SecretsManagerService from '../secretsManagerService';
+
+const app = express();
+const router = express.Router();
+
+const { API_VERSION } = process.env;
+
+// Declare middlewares
+/**
+ * bodyParser, error handling, logger, etc..
+ * http://expressjs.com/en/starter/basic-routing.html
+ * http://expressjs.com/en/guide/using-middleware.html
+ */
+
+/**
+ * app level middlewares
+ * app.use('/path', (req, res, next) => {
+ * chain middlewares
+ * next()
+ * })
+ */
+router.use((req, _response, next) => {
+  // TODO Add logger lib like Winston or Morgan
+  console.log('path');
+  console.log(req.path);
+  next();
+});
+
+// Debug router before we start proxying  requests from /v<x> psth
+router.get('/', (_request, res) => {
+  res.send({ ok: true });
+});
+
+router.get('/version', (_request, res) => {
+  res.send({ version: API_VERSION });
+});
+
+router.get(
+  '/vehicle',
+  (
+    request: Request<Record<string, unknown>, string | Record<string, unknown>, Record<string, unknown>, VehicleEvent>,
+    res,
+  ) => {
+    console.info('Handling vehicle request');
+    let secretsManager;
+    try {
+      secretsManager = new SecretsManagerService(new AWS.SecretsManager());
+    } catch (e) {
+      if (e instanceof Error) {
+        res.status(500).send(e.message);
+      }
+    }
+
+    DatabaseService.build(secretsManager, mysql)
+      .then((dbService) => getVehicleDetails(request.query, vehicleQueryFunctionFactory, dbService))
+      .then((result) => {
+        res.contentType('json').send(result);
+      })
+      .catch((e: Error) => {
+        if (e instanceof ParametersError) {
+          res.status(400);
+        } else {
+          res.status(500);
+        }
+
+        res.send(e.message);
+      });
+  },
+);
+
+router.get(
+  '/testResults',
```

Comment:
I can't see the API specs but I think we use - conventions instead of camel-case for our services. it is probably best we follow that so it is consistent across our services and more intuitive for the users

Reply:
I think this is a @SanjeetBJSS question

Reply:
None of our services are outward facing. I dont mind what we do within the CVS enterprise but I dont like have '_' in API's. It makes them hard to read and mistakes happen. It is just a preference thing.  Can we leave it how it is?

Reply:
If we are all happy it is fine. I was thinking in terms of consistency when a dev tries to use a service without looking at the api specs. if all our api specs when it comes to path use '-' instead of camelCase then it is easier to discover\infer since we use rest. 
For example if we look at the definitions of techrecord paths we have '-' syntax then lambda function call.

```
// config.yml

functions:
- postTestResults:
    method: POST
    path: /test-results
    proxy: null
    function: postTestResults
- getTestResultsBySystemNumber:
    method: GET
    path: /test-results/{+proxy}
    proxy: :systemNumber
    function: getTestResultsBySystemNumber
- getTestResultsByTesterStaffId:
    method: GET
    path: /test-results/getTestResultsByTesterStaffId
    proxy: null
    function: getTestResultsByTesterStaffId
- updateTestResults:
    method: PUT
    path: /test-results/{+proxy}
    proxy: :systemNumber
    function: updateTestResults
   ```